core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -1724,6 +1751,39 @@ object Parser extends ParserInstances {
       null.asInstanceOf[A]
     }
 
+    final def stringIn1[A](radix: RadixNode, all: List[String], state: State): Unit = {
+      val startOffset = state.offset
+      var offset = state.offset
+      var tree = radix
+      var cont = offset < state.str.length
+      var lastMatch = -1
+      while (cont) {
+        val c = state.str.charAt(offset)
+        val idx = Arrays.binarySearch(tree.fsts, c)
+        if (idx >= 0) {
+          val prefix = tree.prefixes(idx)
+          // accept the prefix fo this character
+          if (state.str.startsWith(prefix, offset)) {
+            val children = tree.children(idx)
+            offset += prefix.length
+            tree = children
+            cont = offset < state.str.length
+            if(children.word) lastMatch = offset
+          } else {
+            cont = false
+          }
+        } else {
+          cont = false
+        }
+      }
+      if (lastMatch < 0) {
+        state.error = Chain.one(Expectation.OneStr(startOffset, all))
```

Comment:
I think it would be better to generate the subtree that failed and the offset that we failed at. So, if we have foobar, foobaz, foofoo, and we try to parse fooquux, see would point to quux and say we expected bar, baz or foo. That would be a better error message I think.

What do you think?

Reply:
I like to see the strings as atoms which either match completely or not at all, so I like to see the error at the beginning of the atom. But I am fine either way in the end.

Reply:
I thought about it again, and a typical use case for this operator is to match enum-like names. As a user, I think I am more interested in knowing where it fails at the beginning of the matching of the names than somewhere in the middle of one of the possibilities.

For instance if I want to parse the name of a weekday, I'd have:
```scala
stringIn1(List("monday", "tuesday", "wednesday", "thursday", "firday", "saturday", "sunday"))
```

Let's say I made a type and wrote `thrusday`, it would feel weird to me to know that my possiblities where `hursday` and `uesday`.
Of course this means that the failure offset is closer to the matching error, but semantically speaking, the list in `stringIn1` is what I, as a user, defined as my possibilities, and would expect to see in error messages.

What do you think?

Reply:
I think that makes sense. I guess one thing I'm concerned about is this list getting very big and seeing a giant list on the typo. Like, in your example, I might say, I would expect Thursday or Tuesday, but I don't want to see  Sunday in the list, personally. What do you think of that?